Kickstarter-iOS/Features/ReportProject/ReportProjectFormView.swift

```diff
@@ -8,45 +8,54 @@ enum ReportFormFocusField {
 
 @available(iOS 15.0, *)
 struct ReportProjectFormView: View {
-  let projectID: String
-  let projectURL: String
-  let projectFlaggingKind: GraphAPI.FlaggingKind
   @Binding var popToRoot: Bool
 
   @SwiftUI.Environment(\.dismiss) private var dismiss
-  @ObservedObject private var viewModel = ReportProjectFormViewModel()
+  @ObservedObject private var viewModel: ReportProjectFormViewModel
 
-  @State private var retrievedEmail = ""
-  @State private var details: String = ""
-  @State private var saveEnabled: Bool = false
-  @State private var saveTriggered: Bool = false
   @State private var showLoading: Bool = false
-  @State private var showBannerMessage = false
-  @State private var submitSuccess = false
-  @State private var bannerMessage: MessageBannerViewViewModel?
   @FocusState private var focusField: ReportFormFocusField?
 
+  init(projectID: String,
+       projectURL: String,
+       projectFlaggingKind: GraphAPI.FlaggingKind,
+       popToRoot: Binding<Bool>) {
+    self._popToRoot = popToRoot
+
+    self.viewModel = ReportProjectFormViewModel(
+      projectID: projectID,
+      projectURL: projectURL,
+      projectFlaggingKind: projectFlaggingKind
```

Comment:
Passing these in on init so that `viewModel` has direct access to them. 

One interesting note here - to create a custom init, we needed to include the `Binding<Bool> popToRoot`. The syntax for assigning that binding is weird, hence `self._popToRoot` above.

Reply:
I like that pattern - avoiding the slightly odd custom `init` seems like it would save us some effort.

Reply:
+1. Creating the view model first and then passing it to the view is also nice for testability. I'm not sure if it matters as much in our codebase, since we typically use mocks for testing, but if we ever want to create a fake of an entire view model, we can use that to make simple snapshot tests.